src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/NotificaMittentePagoPATest.java

```diff
@@ -165,6 +165,69 @@ public void nellaSectionInformazioniPreliminariInserireIDatiDellaNotificaSenzaPa
+        switch (variabileAmbiente) {
+            case "dev" -> gruppo = datiNotificaMap.get("gruppoDev");
+            case "test", "uat" -> gruppo = datiNotificaMap.get("gruppoTest");
+        }
+
+        //Sezione preliminare
+        informazioniPreliminariPASection.insertNumeroDiProtocollo(datiNotifica.get("numeroProtocollo").toString());
+        informazioniPreliminariPASection.insertOggettoNotifica(datiNotificaMap.get("oggettoDellaNotifica"));
+        informazioniPreliminariPASection.insertDescrizione(datiNotificaMap.get("descrizione"));
+        WebTool.waitTime(2);
+        informazioniPreliminariPASection.insertGruppo(gruppo);
+        informazioniPreliminariPASection.insertCodiceTassonometrico(datiNotificaMap.get("codiceTassonometrico"));
+        if (datiNotificaMap.get("modello").equals("AR")){
+            informazioniPreliminariPASection.selectRaccomandataAR();
+        }else{
+            informazioniPreliminariPASection.selectRegisteredLetter890();
+        }
+        cliccareSuContinua();
+
+        //Dati destinatario
+        siVisualizzaCorrettamenteLaPaginaPiattaformaNotificheSectionDestinatario();
+        destinatarioPASection.selezionarePersonaFisica();
+        destinatarioPASection.inserireNomeDestinatario(datiNotificaMap.get("nomePF"));
+        destinatarioPASection.inserireCognomeDestinatario(datiNotificaMap.get("cognomePF"));
+        destinatarioPASection.inserireCodiceFiscaleDestinatario(datiNotificaMap.get("codiceFiscalePF"));
+        destinatarioPASection.selezionaAggiungiUnIndirizzoFisico();
+        destinatarioPASection.inserireIndirizzo(datiNotificaMap.get("indirizzoPF"));
+        destinatarioPASection.inserireNumeroCivico(datiNotificaMap.get("numeroCivicoPF"));
+        destinatarioPASection.inserireComune(datiNotificaMap.get("comunePF"));
+        destinatarioPASection.inserireProvincia(datiNotificaMap.get("provinciaPF"));
+        destinatarioPASection.inserireCodicePostale(datiNotificaMap.get("codicepostalePF"));
+        destinatarioPASection.inserireStato(datiNotificaMap.get("statoPF"));
+        destinatarioPASection.vaiInFondoAllaPagina();
+        cliccareSuContinua();
+
+
+        //Sezione allegati
+        siVisualizzaCorrettamenteLaPaginaPiattaformaNotificheSectionAllegati();
+        String pathNotificaFile = notificaFile.getAbsolutePath();
+        allegatiPASection.caricareNotificaPdfDalComputer(pathNotificaFile);
+
+        if (allegatiPASection.verificaCaricamentoNotificaPdf()) {
+            logger.info("File notifica.pdf caricato correttamente");
+        } else {
+            logger.error("File notifica.pdf non caricato");
+            Assert.fail("File notifica.pdf non caricato");
+        }
+        allegatiPASection.inserimentoNomeAllegato(datiNotificaMap.get("nomeDocumentoNotifica"));
+        nellaSectionAllegatiCliccareSulBottoneInvia();
+        siVisualizzaCorrettamenteLaFraseLaNotificaEStataCorrettamenteCreata();
+        cliccareSulBottoneVaiAlleNotifiche();
+    }
```

Comment:
this become a bit too complcated to debug in the future, please keep in mind that you should simplificate the logic by breaking it down into specific methods

Reply:
done